src/Interpreters/InterpreterInsertQuery.cpp

```diff
@@ -391,9 +391,38 @@ Chain InterpreterInsertQuery::buildPreSinkChain(
 
 BlockIO InterpreterInsertQuery::execute()
 {
-    const Settings & settings = getContext()->getSettingsRef();
+    auto context_ptr = getContext();
+    const Settings & settings = context_ptr->getSettingsRef();
     auto & query = query_ptr->as<ASTInsertQuery &>();
 
+    if (query.select)
+    {
+        const ASTSelectWithUnionQuery * ast_select = typeid_cast<ASTSelectWithUnionQuery *>(query.select.get());
```

Comment:
Why did you change it from using `->as<>` to cast?

It's what's used everywhere.

```c++
const auto & select_query = query.select->as<ASTSelectWithUnionQuery &>();
const ASTs & children = select_with_union->list_of_selects->children;
if (children.size() == 1)
...
```

Reply:
updated

Reply:
It has crashed:

```
BaseDaemon: (version 24.1.1.1, build id: 0A7F232DDB35CBB660F88B6DCC0691852FB9112C, git hash: 2092bcbcc91012ad13b29d0759db504b37a806db) (from thread 153887) (query_id: 3c995a91-8ada-47db-b497-8720ffe84f29) (query: create table tab (x String) engine = MergeTree order by x as select 'Hello';) Received signal
```